### ISSUE TITLE
Live reconfiguration - Global

### ISSUE THREAD

**Issue by za419** [za419/CadenceBot#42]
Allow the server admin to perform changes to the loaded configuration from Discord.
This configuration is volatile - It is not saved to disk, and so it will be lost when CadenceBot restarts.

Syntax: One prefix (`Cadence config `) puts us into a configuration flow (thus only needing to check once for admin rights), then the remainder of the command describes what shall be done:

- `set <quotable config key> <quotable value>`
   - This probably isn't JSON, but it could be. Think about that.
   - Custom commands don't need JSON, as each property can be set individually - It's just a little annoying. Perhaps it could be chained?
   - Should this print the old value?
   - Make sure to catch over-nested keys rather than crash (IE allow `Cadence config set unsetkey.child.grandchild true` to create empty keys along the way)
- `get <quotable config key>`
   - Prints JSON result of the key lookup. This has a nice property that a bad key will print as "undefined" without extra work - Thanks, JavaScript!
   - Make sure to catch over-nested keys rather than crash (IE allow `Cadence config get unsetkey.child.grandchild` to print "undefined"
- `delete <quotable config key>`
   - Does exactly what it sounds like - Removes this key from config. For now, we expect only the Very Intelligent (tm) instance administrator to be allowed to use this command, so don't bother checking for safety - If you `Cadence config delete administrator`, you should not be surprised that you can't use admin-only commands, and if you `Cadence config delete customCommands`, you should be smart enough to know that's going to crash the bot next time it sees a non-command message from an unbanned user - For the purposes of this issue, these are the barrier to entry as a CadenceBot instance admin.
- `append <quotable config key> <quotable value>`
   - For use with custom commands, but could be used to bypass the normal ban flow. Can only be used with an array variable (If the target is empty, a primitive, or a non-array object, will fail), and will add the specified value to the array.
   - This is essentially a helper to avoid the need to use set with an index - Removes shall use delete with an index.
- `list`
   - Replaces current `Cadence config` - Dumps out the currently loaded config into the text channel

Examples:

1. `Cadence config set "customCommands.equalTo.Cadence say hello.response" "Hello World!"`
   - Prints `Previously undefined`
   - Afterwards, anyone may send `Cadence say hello` and receive the response `Hello World!`
2. `Cadence config get logging.level`
   - Prints 'logging.level is info'
3. `Cadence config delete "customCommands.targeted.Cadence echo to"`
   - Prints `Deleted, was previously <insert JSON blob here>`
   - Afterwards, `Cadence echo to` will not trigger any response.
4. `Cadence config append "customCommands.Cadence flip a coin.random" Edge...?`
   - Prints `Added one value, array now contains 3 values.`
   - Afterwards, `Cadence flip a coin` will produce the response `Edge...?` with probability ideally one third (RNG-dependent)
5. `Cadence config list`
   - Prints a massive JSON blob, equivalent to running `Cadence config` as of this issue's opening.

**Comment by kenellorando** [comment]
@za419 
~Suggestion: I see use in a `Cadence config reset` command-- something to restore a config file back to a default state (so the bot doesn't have to be restarted to clear any Intelligent configurations). Is this possible?~

**Comment by kenellorando** [comment]
Oh, it's already proposed in #43. Disregard last.

**Comment by za419** [comment]
Yes, the scope for this is such that config changes are potentially very short-lived - Since this is only runnable by the instance admin, and these changes don't save to disk, `reset` isn't really necessary - If I screw up badly enough that I need to restore, it's probably acceptable to just restart the bot.

This is essentially the base, not all that useful, skeleton upon which this whole idea of "Let CadenceBot be personalized to servers!" gets built on.
